app/logic/gossip_node_logic.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module GossipNodeLogic
+  include PipelineLogic
+  include SolanaLogic #for solana_client_request
+
+  def get_nodes
+    lambda do |p|
+      current_nodes = cli_request("gossip", p.payload[:config_urls])
+
+      raise "No nodes returned by cli" unless current_nodes.any?
+
+      Pipeline.new(200, p.payload.merge(current_nodes: current_nodes))
+    rescue StandardError => e
+      Pipeline.new(500, p.payload, "Error from get_nodes", e)
+    end
+  end
+
+  def update_nodes
+    lambda do |p|
+      p.payload[:current_nodes].each do |node|
+        db_node = GossipNode.find_or_create_by(
+          network: p.payload[:network],
+          account: node["identityPubkey"]
+        )
+
+        db_node.ip = node["ipAddress"]
+        db_node.tpu_port = node["tpuPort"]
+        db_node.gossip_port = node["gossipPort"]
+        db_node.version = node["version"]
+
+        db_node.save if db_node.changed?
+
+        db_node.add_validator_ip
+      end
+
+      Pipeline.new(200, p.payload)
+    rescue StandardError => e
+      Pipeline.new(500, p.payload, "Error from update_nodes", e)
+    end
+  end
+
+  def set_staked_flag
+    lambda do |p|
+      staked_validators = Validator.joins(:validator_score_v1)
+                                   .includes(:validator_ips)
+                                   .where("validator_score_v1s.active_stake > 0")
+
+      staked_ips = staked_validators.pluck("validator_ips.address")
+      staked_nodes = GossipNode.where(ip: staked_ips)
+      staked_nodes.update_all(staked: true)
```

Comment:
I found a gossip node without related validator, but yet with `stake == true` flag:
```
irb(main):055:0> GossipNode.find_by(identity: '2t1taw8ye6MdjZkEcia2m7JgVxskrE9pLDkYLJoSWXBD')
=> #<GossipNode id: 2222, identity: "2t1taw8ye6MdjZkEcia2m7JgVxskrE9pLDkYLJoSWXBD", network: "mainnet", ip: "116.202.230.158", tpu_port: nil, gossip_port: 7999, version: "1.10.32", created_at: "2022-08-22 06:48:23.265912000 +0000", updated_at: "2022-08-22 06:48:23.268350000 +0000", staked: true>
irb(main):056:0> Validator.find_by(account: '2t1taw8ye6MdjZkEcia2m7JgVxskrE9pLDkYLJoSWXBD')
=> nil

=> #<GossipNode id: 1810, identity: "CnnjWEyH2eTS2ZGcYojVrbC2eqDnmJgizJMfoLFW9V4N", network: "mainnet", ip: "142.132.135.18", tpu_port: 11003, gossip_port: 11000, version: "1.10.32", created_at: "2022-08-22 06:48:14.614903000 +0000", updated_at: "2022-08-22 06:48:14.616915000 +0000", staked: true>
irb(main):135:0> Validator.find_by(account: 'CnnjWEyH2eTS2ZGcYojVrbC2eqDnmJgizJMfoLFW9V4N')
=> nil

=> #<GossipNode id: 1964, identity: "3TfBxd7UrDq7BvVMgGFMfTWtUqtRNDBJLYGkcw6T1hJF", network: "mainnet", ip: "147.75.109.23", tpu_port: 8004, gossip_port: 8001, version: "1.10.34", created_at: "2022-08-22 06:48:17.219094000 +0000", updated_at: "2022-08-22 06:48:17.221212000 +0000", staked: true>
irb(main):137:0> Validator.find_by(account: '3TfBxd7UrDq7BvVMgGFMfTWtUqtRNDBJLYGkcw6T1hJF')
=> nil

=> #<GossipNode id: 3454, identity: "CoreUyBx7f35cDvtrcZWeymPRWSFoW2XENCGnxKyVWNW", network: "mainnet", ip: "104.244.243.108", tpu_port: 10103, gossip_port: 10100, version: "1.10.35", created_at: "2022-08-22 06:48:53.048362000 +0000", updated_at: "2022-08-22 06:48:53.051087000 +0000", staked: true>
irb(main):140:0> Validator.find_by(account: 'CoreUyBx7f35cDvtrcZWeymPRWSFoW2XENCGnxKyVWNW')
=> nil

=> #<GossipNode id: 3171, identity: "6SbfJRFMoj2YwpKzzvxyraaj25Liyd3r8yU74z6SQ5Lq", network: "mainnet", ip: "69.197.2.5", tpu_port: nil, gossip_port: 8001, version: "1.10.34", created_at: "2022-08-22 06:48:46.456821000 +0000", updated_at: "2022-08-22 06:48:46.459069000 +0000", staked: true>
irb(main):142:0> Validator.find_by(account: '6SbfJRFMoj2YwpKzzvxyraaj25Liyd3r8yU74z6SQ5Lq')
=> nil


```

Is it all correct in this query? also looks like it is not covered in tests enough since that case could happen.